src/WebJobs.Script/Binding/ServiceBusScriptBindingProvider.cs

```diff
@@ -110,8 +112,17 @@ public override Type DefaultType
                 {
                     if (Context.Access == FileAccess.Read)
                     {
-                        return string.Compare("binary", Context.DataType, StringComparison.OrdinalIgnoreCase) == 0
+                        Type type = string.Compare("binary", Context.DataType, StringComparison.OrdinalIgnoreCase) == 0
                             ? typeof(byte[]) : typeof(string);
+
+                        if (string.Compare("many", Context.Cardinality, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            // arrays are supported for both trigger input as well
+                            // as output bindings
+                            type = type.MakeArrayType();
+                        }
```

Comment:
This logic is extension specific. E.g. another extension might not support an array binding, and instead map "many" to a different collection type. How can I query binding rules to discover these mappings? I don't think that is possible currently.

Let's track any changes to this model as a separate issue, since here I'm following the existing pattern. I'm all for simplifying if we can.